File: greedy/maximum-matching-of-players-with-trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        sort(players.begin(), players.end());
        sort(trainers.begin(), trainers.end());
        
        int l = 0, r = 0, ans = 0;
        while (l < players.size() && r < trainers.size()) {
            if (players[l] <= trainers[r]) {  
                ans++;
                l++; 
            }
            r++;  
        }
        return ans;
    }
};

File: array/maximum-matching-of-players-with-trainers.java
class Solution {
    public int matchPlayersAndTrainers(int[] players, int[] trainers) {
        Arrays.sort(players);
        Arrays.sort(trainers);
        int l = 0, r = 0, ans = 0;
        for (; l < players.length; l++) {
            if (r < trainers.length && players[l] <= trainers[r]) {
                r++;
                ans++;
            } else {
                while (r < trainers.length && players[l] > trainers[r]) {
                    r++;
                }
                if (r == trainers.length) break;
                else ans++;
            }
        }
        return ans;
    }
}